File: problems/1614_maximum_nesting_depth_of_the_parentheses/csharp/1614_maximum_nesting_depth_of_the_parentheses.cs
using System;

public static class Program
{
  public static void Main(string[] args)
  {
    Solution s = new Solution();
    Console.WriteLine(s.MaxDepth("(1+(2*3)+((8)/4))+1"));
    Console.WriteLine(s.MaxDepth("(1)+((2))+(((3)))"));
    Console.WriteLine(s.MaxDepth("()(())((()()))"));
  }
}

public class Solution
{
  public int MaxDepth(string s)
  {
    int max = 0;
    int depth = 0;
    foreach (char c in s)
    {
      switch (c)
      {
        case '(': depth++; break;
        case ')': depth--; break;
      }
      if (depth > max) max = depth;
    }
    return max;
  }
}

File: problems/1614-maximum-nesting-depth-of-the-parentheses/python/1614_maximum_nesting_depth_of_the_parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        max: int = 0
        depth: int = 0
        for c in s:
            if c == '(':
                depth += 1
            elif c == ")":
                depth -= 1
            if depth > max:
                max = depth
        return max


def test(s: str) -> None:
    result: int = Solution().maxDepth(s)
    print(f"Max depth for these parantheses is {
          result}!\n -> parantheses: {s}")


if __name__ == "__main__":
    test("(1+(2*3)+((8)/4))+1")
    test("(1)+((2))+(((3)))")
    test("()(())((()()))")
